### PR TITLE
Chainmail doc

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Quick writeup on context and usage of chainmail.

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
https://github.com/Agoric/ERTP/issues/126
https://github.com/Agoric/ERTP/issues/14

**Comment by katelynsills** [comment]
It looks like this version is out-of-date with master, probably because of recent commits there

**Comment by Chris-Hibbert** [comment]
fixed

File: CHAINMAIL.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+The work with Victoria University
+[Reasoning about Risk and Trust in an Open World](https://ai.google/research/pubs/pub44272) has
+resulted in development of a specification language, Chainmail, which is intended to allow programs
+to be described with enough formality to investigate their security properties. We have used a
+variant of the language to describe some of the core modules of
+ERTP. [assays](https://github.com/Agoric/ERTP/blob/master/core/assays.chainmail),
+[contractHost](https://github.com/Agoric/ERTP/blob/master/core/contractHost.chainmail),
+[issuers](https://github.com/Agoric/ERTP/blob/master/core/issuers.chainmail).
+
+We have several ideas for making use of chainmail descriptions in addition to formal proofs:
```
Comment (erights) [inline CHAINMAIL.md]: This sounds like it's claiming too much. We are only using an IDL-only subset of chainmail, with the intention of expanding it eventually for formal specification. What is in there right now does no more formal specification than any other IDL.

Comment (Chris-Hibbert) [reply CHAINMAIL.md]: done.

File: CHAINMAIL.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+Part of the Agoric strategy for providing security is based on the
+[Formal Methods](https://agoric.com/papers/#formal-reasoning) literature, which has developed tools
+for proving that programs follow their specifications. There have been interesting advances in this
+area including the development of [seL4](https://sel4.systems/Info/FAQ/proof.pml) and the
+[proof of its correctness](https://sel4.systems/Info/FAQ/proof.pml). We have been collaborating with
+a research group at Victoria University who are investigating how to apply these techniques to
```
Comment (erights) [inline CHAINMAIL.md]: The group is actually split between Imperial (Sophia and Susan), Victoria (James), and Data61 (Toby). See the attributions at the top of https://ai.google/research/pubs/pub44272.pdf but add Susan Eisenbach of Imperial

File: CHAINMAIL.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+secure distributed programs.
+
+The work with Victoria University
+[Reasoning about Risk and Trust in an Open World](https://ai.google/research/pubs/pub44272) has
+resulted in development of a specification language, Chainmail, which is intended to allow programs
+to be described with enough formality to investigate their security properties. We have used a
```
Comment (erights) [inline CHAINMAIL.md]: Still sounds like we're claiming more than our current chainmail provides. Make explicit that it is only a subset that corresponds to a conventional object IDL, and that it starts with a syntactic subset of the Cap'n Proto IDL.

File: CHAINMAIL.md (hunk 2 above)
Comment (Chris-Hibbert) [reply CHAINMAIL.md]: I just listed institutions. I gave UNSW for Toby. Would individual names be better?

File: CHAINMAIL.md (hunk 3 above)
Comment (Chris-Hibbert) [reply CHAINMAIL.md]: Done.

File: CHAINMAIL.md (hunk 2 above)
Comment (erights) [reply CHAINMAIL.md]: > Would individual names be better?

yes

Comment (Chris-Hibbert) [reply CHAINMAIL.md]: done

**Comment by katelynsills** [comment]
I'm good to merge this in!

**Comment by Chris-Hibbert** [comment]
Approved by both, with requested changes.

**Pull request merged**